tests/spec/component_spec.lua

```diff
@@ -258,14 +258,21 @@ end)
 
 describe('Filetype component', function()
   local filetype
+  local orig_expand
 
   before_each(function()
     filetype = vim.bo.filetype
     vim.bo.filetype = 'lua'
+    orig_expand = vim.fn.expand
+    vim.fn.expand = function(fmt)
```

Comment:
You can use `stubs` from testing framework instead of overwriting the function yourself. There are examples in the test suit and you can take a look at the lua busted docs for more info.

Reply:
While at it shall I use `stub` for accessing `nvim-web-devicons` as well? If yes, should I put it on another (refactoring) PR to focus on file type icons in this one?

Reply:
Sure if you feel like it you can do that too.